tritium-registry/src/main/java/com/palantir/tritium/metrics/registry/TaggedMetricRegistry.java

```diff
@@ -144,6 +145,8 @@ default void registerWithReplacement(MetricName metricName, Gauge<?> gauge) {
      */
     void addMetrics(@Safe String safeTagName, @Safe String safeTagValue, TaggedMetricSet metrics);
 
+    void addMetrics(Map<@Safe String, @Safe String> tags, TaggedMetricSet metrics);
```

Comment:
Would this mean I'd be able to specify 0 tags? Cause that combined with metric-schema being able to group metrics by tags, would be my preferred solution. And actually something like:

```
interface Closer {
}

Closer addMetrics(Map<String, String> tags, TaggedMetricSet metrics);
```

is probably my favourite of all.

Reply:
Can you describe cases where you'd want an empty tag map? I don't think we've touched this API in forever, so need to look at how consumers are using it/want to use it a bit more.

I'm not sure I quite follow the `Closer` proposal. Can you describe the use case/more pseudocode example? Are you envisioning use in try-with-resources to perform some cleanup callback?

Reply:
Sorry, should've fleshed it out.

> Can you describe cases where you'd want an empty tag map?
See the internal PR that sparked this conversation. Effectively adding a scoped TaggedMetricRegistry, if metric-schema supports adding per-namespace tags, means that ALL tags that apply to a metric set ARE documented using standard tools (i.e. by adding them in the "tags" section of the yaml), as opposed to in the "documentation" field which is free-form. Right now, if you have a look at internal skiing database or the PR I mentioned, when using the child TaggedMetricRegistries, you end up documenting it in a free-form way.

> I'm not sure I quite follow the Closer proposal. Can you describe the use case/more pseudocode example? Are you envisioning use in try-with-resources to perform some cleanup callback?

Yes, exactly. Being able to use try-with-resources would be useful in the context of the PR which sparked this PR. But it effectively removes the foot-gun of accidentally calling add followed by remove with the tags mixed up somehow.

Reply:
@carterkozak well that's another thing which I think we should ideally solve: the other perverse case is where we actually report things for a while, and then it gets removed and we don't report the final set of values, which is also not ideal.

Reply:
I think I have another reason why we want the ability to not specify any tags on a child metric registry: our internal 'titan's child' product is not able to validate tags not defined in metric-schema. Meaning if anyone ever needs to write a check using the 'titan's child' product they will not be able to use the tags passed to #addMetrics method.

Reply:
Hmm maybe I'm wrong and there's a special feature for this...

Reply:
Anyway, @carterkozak suggested just pushing a useless tag that doesn't mean anything, I think that's good enough

Reply:
At this point the Closer idea is probably a nice to have.